Upgrades to support node 10

DullReferenceException opened the pull request:
* Require `gulp@4` since `gulp@3` has node-10-incompatible dependencies
* Since `gulp@4` has breaking changes, use the newer task export pattern
* Remove `gulp-help`; author is not publishing a `gulp@4` update
* Remove deprecated `gulp-util` and replace with recommendation
* Upgrade to a `gulp@4`-compatible version of `gulp-sonar`
* Fix `npm audit` issues

jpage-godaddy committed:
Upgrades to support node 10

* Require `gulp@4` since `gulp@3` has node-10-incompatible dependencies
* Since `gulp@4` has breaking changes, use the newer task export pattern
* Remove `gulp-help`; author is not publishing a `gulp@4` update
* Remove deprecated `gulp-util` and replace with recommendation
* Upgrade to a `gulp@4`-compatible version of `gulp-sonar`
* Fix `npm audit` issues

SivanMehta commented:
You should add a [`.travis.yml`](https://github.com/godaddy/warehouse.ai/blob/master/.travis.yml#L4-L5) that tests this against both node@8 and node@10 automatically.

DullReferenceException commented:
@SivanMehta, unfortunately this repo doesn't have any tests.

jpage-godaddy committed:
Fix glob

Although the glob bump was supposed to be semver compatible, it now
errors out if a glob pattern does not match a file.

* Update `gulp.src` calls to allow empty glob matches
* Fix gulpfile for this project

DullReferenceException commented:
Closing; my changes were merged in a different PR.
